src/main/java/org/snomed/snowstorm/core/data/services/CodeSystemService.java

```diff
@@ -484,6 +495,14 @@ public CodeSystemVersion findLatestVisibleVersion(String shortName) {
 		return null;
 	}
 
+	public CodeSystemVersion findPreviousVisibleVersion(String shortName) {
+		List<CodeSystemVersion> versions = findAllVersions(shortName, false, latestVersionCanBeFuture);
+		if (versions != null && versions.size() > 1) {
```

Comment:
I have set the condition to be greater than 1 as there were some Code Systems (LOINC for example) that only had 1. Therefore, if a Code System doesn't have more than 1, then return `null` as that essentially indicates there is no previous version.